src/models/output.ts

```diff
@@ -210,6 +233,44 @@ class Output {
 
     return [output, outputBuffer];
   }
+
+  /**
+   * Checks if the script length is within the valid limits
+   *
+   * @returns {boolean} True if the script is within valid limits
+   *
+   * @memberof Output
+   * @inner
+   */
+  hasValidLength(): boolean {
+    // No script can have more than the maximum length
+    if (this.script.length > MAXIMUM_SCRIPT_LENGTH) {
+      return false;
+    }
+
+    // P2PKH and P2SH scripts can only have 30 or 25 bytes (with or without timelock)
+    if (P2PKH.identify(this.script) || P2SH.identify(this.script)) {
+      if (!(this.script.length === 25 || this.script.length === 30)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Returns the type of the output, according to the specified network
+   *
+   * @param {Network} network Network to get output addresses first byte
+   * @returns {string} Output type
+   *
+   * @memberof Output
+   * @inner
+   */
+  getType(network: Network): String|undefined {
+    const decodedScript = this.decodedScript || this.parseScript(network)
+    return decodedScript?.getType();
```

Comment:
```js
    return decodedScript?.getType() || null;
    // or
    return decodedScript?.getType() || '';
```
I believe it's better to return `null` than `undefined`, but the empty string could be a better option since it keeps the type consistent.

Reply:
Done on [e095f56](https://github.com/HathorNetwork/hathor-wallet-lib/pull/378/commits/e095f560dc99bf237f230df3579027edb4f91870).